docs/src/index.md

```diff
@@ -181,8 +181,8 @@ linear (generic function with 2 methods)
 julia> grads = gradient(() -> sum(linear(x)), Params([W, b]))
 Grads(...)
 
-julia> grads[W], grads[b]
+julia> grads[W], grads[b] # access gradients using arrays as keys
 ([0.652543 … 0.683588], [1.0, 1.0])
 ```
 
-However, implicit parameters exist mainly for compatibility with Flux's current AD; it's recommended to use the other approaches unless you need this.
+Here `grads` is a dictionary-like object, whose keys are the same parameters we indicated in `Params`. (In fact it wraps a dictionary using `objectid(W)` as keys, which does not change if the values in `W` are mutated).
```

Comment:
This is the paragraph which is migrating downwards again:
```suggestion
Here `grads` is a dictionary-like object, whose keys are the same parameters we indicated in `Params`. (In fact it wraps a dictionary using `objectid(W)` as keys, which does not change if the values in `W` are mutated).

This implicit style is the one presently used by [Flux.jl](https://github.com/FluxML/Flux.jl), a closely related machine learning library. It uses structs like `Linear` above to define layers, and the function `Flux.params(model)` returns a `Params` object containing all the parameters of all layers. See [its documentation](https://fluxml.ai/Flux.jl/stable/models/basics/) for more details. When using Zygote for most other purposes, however, the explicit style is usually preferred.
```

Also todo, I think it should not say "Flux-style model" above. 

And I wonder whether having (and referring to) two different functions called `linear` is such a bright idea. Maybe the 2nd one should be `linearWb(x)`?